### PR TITLE
refactor(examples): use consistent naming

### LINKED ISSUE #4459: examples: use consistent naming
## Description

We have several examples in our repository in https://github.com/libp2p/rust-libp2p/tree/master/examples. Examples are often the first thing users get in touch with so they should be consistent and easy to understand.

Some of these examples are named `-example` where as others are not. I am suggesting that we:

- Remove all `-example` postfixes from the directory names
- Add a `-example` to all _crate_ names to ensure that all the binaries have a descriptive name

## Motivation

Consistent code makes things easier.

## Current Implementation

A mix between `-example` and no postfix.

### PR DESCRIPTION
**Pull request by Blaeaea**
## Description

Resolves: #4459.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -92,7 +92,7 @@ libp2p-relay = { version = "0.16.1", path = "protocols/relay" }
 libp2p-rendezvous = { version = "0.13.0", path = "protocols/rendezvous" }
 libp2p-request-response = { version = "0.25.1", path = "protocols/request-response" }
 libp2p-server = { version = "0.12.2", path = "misc/server" }
-libp2p-swarm = { version = "0.43.4", path = "swarm" }
+libp2p-swarm = { version = "0.43.3", path = "swarm" }
```
Comment (thomaseizinger) [inline Cargo.toml]: This seems like an unrelated change?

**Review by thomaseizinger — COMMENTED** [review]
Thank you!

In order to complete #4459, I'd also like to have all the crate names adjusted, i.e. the `name = XYZ` lines in the various `Cargo.toml`s.

By convention, cargo will use those to name the binaries produced by the examples. Hence, those crate names should have `-example` make it clear during compilation, what is compiled and to not clash with any other binaries we might produce (as part of tests for example).

Can you address this too please?

**Comment by Blaeaea** [comment]
> Thank you!
> 
> In order to complete #4459, I'd also like to have all the crate names adjusted, i.e. the `name = XYZ` lines in the various `Cargo.toml`s.
> 
> By convention, cargo will use those to name the binaries produced by the examples. Hence, those crate names should have `-example` make it clear during compilation, what is compiled and to not clash with any other binaries we might produce (as part of tests for example).
> 
> Can you address this too please?

Hello, I apologize for any confusion. Could you please provide some examples to illustrate what needs to be done? Thank you!

**Comment by thomaseizinger** [comment]
> > Thank you!
> > In order to complete #4459, I'd also like to have all the crate names adjusted, i.e. the `name = XYZ` lines in the various `Cargo.toml`s.
> > By convention, cargo will use those to name the binaries produced by the examples. Hence, those crate names should have `-example` make it clear during compilation, what is compiled and to not clash with any other binaries we might produce (as part of tests for example).
> > Can you address this too please?
> 
> Hello, I apologize for any confusion. Could you please provide some examples to illustrate what needs to be done? Thank you!

I can't push to your branch unfortunately but here is an example patch:

```patch
Subject: [PATCH] Rename example to be consistent
---
Index: Cargo.lock
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Cargo.lock b/Cargo.lock
--- a/Cargo.lock	(revision 2274da0d5eb828952849942dddb8c8909fef4579)
+++ b/Cargo.lock	(revision 4aea220d59e5fbde766e257ef739efddc06ae10d)
@@ -1314,7 +1314,7 @@
 ]
 
 [[package]]
-name = "dcutr"
+name = "dcutr-example"
 version = "0.1.0"
 dependencies = [
  "clap",
Index: examples/dcutr/Cargo.toml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/examples/dcutr/Cargo.toml b/examples/dcutr/Cargo.toml
--- a/examples/dcutr/Cargo.toml	(revision 2274da0d5eb828952849942dddb8c8909fef4579)
+++ b/examples/dcutr/Cargo.toml	(revision 4aea220d59e5fbde766e257ef739efddc06ae10d)
@@ -1,5 +1,5 @@
 [package]
-name = "dcutr"
+name = "dcutr-example"
 version = "0.1.0"
 edition = "2021"
 publish = false
```

Doing this for all examples is what I'd like. Most importantly, this will also rename all binaries produced by the example. As you can see from https://github.com/libp2p/rust-libp2p/pull/4458, some of these binaries are used in the docs of other examples.

Once you've renamed all the examples, can you please go through all docs of other examples and check that all the commands still work? Thank you! :)
